### PR TITLE
Implant Tune-up

### PR DESCRIPTION
**Pull request by ghost**
This PR does a lot of cleaning of implant code (such as changing the god awful variable name of imp_in) and adds new sprites for the implanter. It also adds a implantvend that I didn't add into the map yet. Why? Well, after finishing the sprite I realized that I'm not even sure why Shifty asked me to make it, nor could I figure out a reason to have loyalty implants in a vendor. So it's in the code. 
The new implanter skin changes color based on what's in it, default color green.

implant_vendor_big

### CODE REVIEW

**Comment by ShiftyRail** [comment]
Might be gunky but I'd rather have all implants in it rather than on a rack on the armoury

File: code/modules/implants/implantvend.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	desc = "A vending machine filled with a wide variety of implants."
+	req_access = list(access_security)
+	product_ads = list(
+		"Motivate! Motivate! Motivate!",
+		"The good stuff is in the back.",
+		"Ahem. Fuck revolutionaries, and fuck the occult!",
+		"Loyalty implants will pay themselves off in increased productivity!",
+		"I'm sure you've got lots of spare cash! Put it in!",
+		"There doesn't need to be subversive elements for you to implant!",
+		"Become a beast today! Fill your body with perfectly legal amounts of adrenalin!"
```
Comment (juanototo2) [inline code/modules/implants/implantvend.dm]: 🤔

**Comment by juanototo2** [comment]
ignore that review, I don't know how to github

File: code/modules/implants/implantvend.dm [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/obj/machinery/vending/implant
+	name = "\improper ImplantVend"
+	desc = "A vending machine filled with a wide variety of implants."
+	req_access = list(access_security)
+	product_ads = list(
+		"Motivate! Motivate! Motivate!",
+		"The good stuff is in the back.",
+		"Ahem. Fuck revolutionaries, and fuck the occult!",
```
Comment (juanototo2) [inline code/modules/implants/implantvend.dm]: :thinking

Comment (ShiftyRail) [reply code/modules/implants/implantvend.dm:8]: Dare I say based

File: code/modules/implants/implantcase.dm [hunk 3]
```diff
@@ -0,0 +1,103 @@
+/obj/item/weapon/implantcase
+	name = "Glass Case"
```
Comment (Kurfursten) [inline code/modules/implants/implantcase.dm:2]: not a proper noun

File: code/modules/implants/implantcase.dm [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+	item_state = "implantcase"
+	throw_speed = 1
+	throw_range = 5
+	w_class = W_CLASS_TINY
+	var/obj/item/weapon/implant/held_implant = null
+	var/implant_path = null
+	
+/obj/item/weapon/implantcase/New()
+	..()
+	src.held_implant = new implant_path(src)
```
Comment (Kurfursten) [inline code/modules/implants/implantcase.dm:14]: the many srcs in this file are unnecessary

File: code/modules/implants/implantcase.dm [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+	desc = "A case containing a chemical implant."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "implantcase-blue"
+	implant_path = /obj/item/weapon/implant/chem
+
+/obj/item/weapon/implantcase/loyalty
+	name = "Glass Case- 'Loyalty'"
+	desc = "A case containing a loyalty implant."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "implantcase-red"
```
Comment (Kurfursten) [inline code/modules/implants/implantcase.dm:82]: I feel like if anything's gonna use classic green, loyalty should be it, due to the HUD element being a green square

File: code/modules/virus2/items_devices.dm [hunk 6]
```diff
@@ -26,7 +26,7 @@
 /obj/item/weapon/virusdish
 	name = "virus containment/growth dish"
 	icon = 'icons/obj/items.dmi'
-	icon_state = "implantcase-b"
+	icon_state = "implantcase-blue"
```
Comment (Kurfursten) [inline code/modules/virus2/items_devices.dm:29]: people probably are not expecting their virus dishes to change color

Comment (ghost) [reply code/modules/virus2/items_devices.dm:29]: those are the same, I renamed it for clarity

File: code/modules/implants/implantcase.dm (hunk 5 above)
Comment (ghost) [reply code/modules/implants/implantcase.dm:82]: Good point. Will change to blue, there's no green sprite.

File: code/modules/implants/implantcase.dm (hunk 4 above)
Comment (ghost) [reply code/modules/implants/implantcase.dm:14]: Copied from old implant code, will condense some of them.

File: code/modules/implants/implantcase.dm (hunk 3 above)
Comment (ghost) [reply code/modules/implants/implantcase.dm:2]: you're not a proper noun
"kurfursten"

Comment (Kurfursten) [reply code/modules/implants/implantcase.dm:2]: Too bad for you that our repo is not in German, where all nouns are capitalized.

Comment (PaulRitter) [reply code/modules/implants/implantcase.dm:2]: *dabs in german

**Comment by Kurfursten** [comment]
do that stuff

**Pull request closed without merging**